tests/integration/warp_util.go

```diff
@@ -0,0 +1,272 @@
+package integration
+
+import (
+	"cosmossdk.io/math"
+
+	ismTypes "github.com/bcp-innovations/hyperlane-cosmos/x/core/01_interchain_security/types"
+	pdTypes "github.com/bcp-innovations/hyperlane-cosmos/x/core/02_post_dispatch/types"
+	coreKeeper "github.com/bcp-innovations/hyperlane-cosmos/x/core/keeper"
+	coreTypes "github.com/bcp-innovations/hyperlane-cosmos/x/core/types"
+	"github.com/bcp-innovations/hyperlane-cosmos/x/middleware"
+	"github.com/bcp-innovations/hyperlane-cosmos/x/warp"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/bcp-innovations/hyperlane-cosmos/util"
+	"github.com/bcp-innovations/hyperlane-cosmos/x/warp/keeper"
+	"github.com/bcp-innovations/hyperlane-cosmos/x/warp/types"
+	"github.com/cosmos/gogoproto/proto"
+
+	. "github.com/onsi/gomega"
+)
+
+var denom = "acoin"
+
+func (suite *KeeperTestSuite) setupWarpApps() {
+	defaultWarpApps := warp.DefaultWarpApps(&suite.app.WarpKeeper)
+
+	// If no custom hooks, use default registration
+	if len(suite.HandleHooks) == 0 {
+		suite.app.RegisterWarpApps(defaultWarpApps...)
+		return
+	}
+
+	warpAppsMap := make(map[types.HypTokenType]util.HyperlaneApp, len(defaultWarpApps))
+	for _, app := range defaultWarpApps {
+		warpAppsMap[app.TokenType] = app.App
+	}
+
+	// Wrap app with custom middleware if hooks are provided.
+	for tokenType, hook := range suite.HandleHooks {
+		app, ok := warpAppsMap[tokenType]
+		Expect(ok).To(BeTrue())
+
+		appMiddleware, err := middleware.NewMiddleware(app, hook)
+		Expect(err).To(BeNil())
+
+		warpAppsMap[tokenType] = appMiddleware
+	}
+
+	warpApps := make([]warp.WarpApp, 0, len(warpAppsMap))
+	for tokenType, app := range warpAppsMap {
+		warpApps = append(warpApps, warp.WarpApp{
+			TokenType: tokenType,
+			App:       app,
+		})
+	}
+
+	suite.app.RegisterWarpApps(warpApps...)
+}
+
+func createIgp(s *KeeperTestSuite, creator string) util.HexAddress {
+	res, err := s.RunTx(&pdTypes.MsgCreateIgp{
+		Owner: creator,
+		Denom: denom,
+	})
+	Expect(err).To(BeNil())
+
+	var response pdTypes.MsgCreateIgpResponse
+	err = proto.Unmarshal(res.MsgResponses[0].Value, &response)
+	Expect(err).To(BeNil())
+
+	return response.Id
+}
+
+func createMerkleHook(s *KeeperTestSuite, creator string, mailboxId util.HexAddress) util.HexAddress {
+	res, err := s.RunTx(&pdTypes.MsgCreateMerkleTreeHook{
+		Owner:     creator,
+		MailboxId: mailboxId,
+	})
+	Expect(err).To(BeNil())
+
+	var response pdTypes.MsgCreateMerkleTreeHookResponse
+	err = proto.Unmarshal(res.MsgResponses[0].Value, &response)
+	Expect(err).To(BeNil())
+
+	return response.Id
+}
+
+func CreateValidMailbox(s *KeeperTestSuite, creator string, ism string, destinationDomain uint32) (util.HexAddress, util.HexAddress, util.HexAddress) {
+	var ismId util.HexAddress
+	switch ism {
+	case "noop":
+		ismId = CreateNoopIsm(s, creator)
+	case "multisig":
+		ismId = createMultisigIsm(s, creator)
+	}
```

Comment:
nit: handle default here


```suggestion
	switch ism {
	case "noop":
		ismId = CreateNoopIsm(s, creator)
	case "multisig":
		ismId = createMultisigIsm(s, creator)
    default:
        Expect(true).To(BeFalse(), fmt.Sprintf("unexpected ism type: %q", ism))
	}
```